docs/reference/settings.md

```diff
@@ -1696,6 +1696,34 @@ Only applies to `pyproject.toml`, `setup.py`, and `setup.cfg` sources.
 
 ---
 
+#### [`all-groups`](#pip_all-groups) {: #pip_all-groups }
+<span id="all-groups"></span>
+
+Include all groups.
+
+Only applies to `pyproject.toml`, `setup.py`, and `setup.cfg` sources.
+
+**Default value**: `false`
+
+**Type**: `bool`
+
+**Example usage**:
+
+=== "pyproject.toml"
+
+    ```toml
+    [tool.uv.pip]
+    all-groups = true
+    ```
+=== "uv.toml"
+
+    ```toml
+    [pip]
+    all-groups = true
+    ```
+
+---
+
```

Comment:
Honestly I'm not sure why we support requesting extras in settings file like this. It seems correct to match the extras options for groups though?

Perhaps @charliermarsh understands the use-case?